compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertAIEWorkgroup.cpp

```diff
@@ -152,6 +167,10 @@ class AMDAIEInsertAIEWorkgroupPass
 };
 
 void AMDAIEInsertAIEWorkgroupPass::runOnOperation() {
+  // Normalize the loop bounds of `scf.forall` operations within the module.
+  if (failed(normalizeModuleLoopBounds(getOperation()))) {
+    return signalPassFailure();
+  }
```

Comment:
Any reason why we're doing it this way ?

I guess this might not be the right way to do things.

I'd instead have the normalisation pass invoked before `AMDAIEInsertAIEWorkgroupPass`. This also helps during debugging. Else by clubbing such major changes within one pass, we are missing out on lot of ways which we can use during debugging an IR log.

I see a similar thing being done in [AMDAIEUnrollAndDistributeWorkgroupPass here](https://github.com/nod-ai/iree-amd-aie/blob/main/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUnrollAndDistributeWorkgroup.cpp#L467-L499). I faced the above issue today while debugging the pass hence thought it'd actually make sense to break down passes like these into smaller chunks that does one thing atomically. Just helps getting a good look of IR transformations.

Reply:
Initially, I walked the workgroups here and only normalized the loops within the workgroups, but now, with normalizing the whole module, I agree we should just call the pass before. 

In general, is there a way to add dependency passes for a pass that are always run before/after the corresponding pass in a pipeline?

> I see a similar thing being done in [AMDAIEUnrollAndDistributeWorkgroupPass here](https://github.com/nod-ai/iree-amd-aie/blob/main/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEUnrollAndDistributeWorkgroup.cpp#L467-L499). I faced the above issue today while debugging the pass hence thought it'd actually make sense to break down passes like these into smaller chunks that does one thing atomically. Just helps getting a good look of IR transformations.

Not sure about the `AMDAIEUnrollAndDistributeWorkgroupPass` as I think it's quite atomic, just with a couple of back-to-back steps. For example, splitting between `AMDAIEUnrollWorkgroupLoops` and `assignLocalAieTiles` actually doesn't give correct results if you would run `cse` in between. Maybe, you would get out the `workgroupForallToFor` and `hoistAffineApplyDependingOnFor` out of it, but I think it's quite specific to the unroll that follows?

As for debuggability, we can add debug statements in between the different steps that dump the IR in between?